File: Reqnroll/Chapter13/13.3-TheIntroduceAndReferPattern/After/WIMP.IntroduceAndReferSample.Specs/StepDefinitions/AuthenticationStepDefinitions.cs
using Reqnroll;

using WIMP.IntroduceAndReferSample.App.Services;
using WIMP.IntroduceAndReferSample.Specs.Support;

namespace WIMP.IntroduceAndReferSample.Specs.StepDefinitions;

[Binding]
public class AuthenticationStepDefinitions(AuthenticationContext authContext)
{
    [Given("the customer {word} has logged in")]
    public void GivenTheCustomerHasLoggedIn(string customerName)
    {
        AuthenticationService.Login(customerName);
        authContext.LoggedInCustomerName = customerName;
    }
}

File: CucumberJVM/Chapter13/13.3-IntroduceAndReferPattern/Before/src/test/java/com/wimp/app/specs/stepdefinitions/AuthenticationStepDefinitions.java
package com.wimp.app.specs.stepdefinitions;

import com.wimp.app.services.AuthenticationService;
import com.wimp.app.specs.support.AuthenticationContext;
import io.cucumber.java.en.*;

public class AuthenticationStepDefinitions {
    private final AuthenticationContext authenticationContext;

    public AuthenticationStepDefinitions(AuthenticationContext authenticationContext) {
        this.authenticationContext = authenticationContext;
    }

    @Given("the customer {string} is authenticated")
    public void theCustomerIsAuthenticated(String customerName) {
        AuthenticationService.login(customerName);
        authenticationContext.setAuthenticatedCustomerName(customerName);
    }
}
